### PR TITLE
Explore using Github Actions for CI

### PR DESCRIPTION
**Pull request by czoido**
Changelog: (Feature | Fix | Bugfix): Describe here your pull request
Docs: https://github.com/conan-io/docs/pull/XXXX

Just the basics for Linux and unittests

cc/ 
@franramirez688 --> Windows
@uilianries, @perseoGI --> macOS

### CODE REVIEW

**Comment by memsharded** [comment]
Previously we were only building pre-merge commit, isn't it? 
This would be building both the pushed commit and the merge-commit?

**Review by memsharded — APPROVED** [review]
Looking great

**Comment by czoido** [comment]
> Previously we were only building pre-merge commit, isn't it? This would be building both the pushed commit and the merge-commit?

I think right now in Jenkins for PR's we are only building the _virtual_ merge commit.
For this PR with GitHub actions we build against the _virtual_ merge commit for the PR and also the pushed commit of the branch (pre-merge) but we can change it to only do the merged commit like we did in Jenkins.

**Comment by memsharded** [comment]
> For this PR with GitHub actions we build against the virtual merge commit for the PR and also the pushed commit of the branch (pre-merge) but we can change it to only do the merged commit like we did in Jenkins.

I think I prefer to have just 1 thing to check, the merged commit looks good to me, but I am fine if others think it has value.
At the end of the day, the recommendation is to be as updated from the target branch as possible, so I have never seen difference in the result of the merge commit or the pre-merge commit.

**Comment by czoido** [comment]
> > For this PR with GitHub actions we build against the virtual merge commit for the PR and also the pushed commit of the branch (pre-merge) but we can change it to only do the merged commit like we did in Jenkins.
> 
> I think I prefer to have just 1 thing to check, the merged commit looks good to me, but I am fine if others think it has value. At the end of the day, the recommendation is to be as updated from the target branch as possible, so I have never seen difference in the result of the merge commit or the pre-merge commit.

Yes, I agree, better to just check the merged commit and push commits only on develop2 and release branches.

**Comment by uilianries** [comment]
I talked to Perseo on Friday, so we can continue together the Mac support. I did few tests on my fork, so far, I can run the unit tests: https://github.com/uilianries/conan/blob/ci/gh-osx/.github/workflows/osx-tests.yml

I still need to investigate the separation for running on Intel and M1. I know GH offers separated images for them. And also, configure/install different versions of SDK.

**Comment by czoido** [comment]
> I talked to Perseo on Friday, so we can continue together the Mac support. I did few tests on my fork, so far, I can run the unit tests: https://github.com/uilianries/conan/blob/ci/gh-osx/.github/workflows/osx-tests.yml
> 
> I still need to investigate the separation for running on Intel and M1. I know GH offers separated images for them. And also, configure/install different versions of SDK.

I think it's ok to test only on M1, in fact, we are only testing on M2 right now in the ci, I would keep things as simple as possible for the moment.

File: ci/conan-tests
Comment (valgur) [inline ci/conan-tests:1]: This Dockerfile looks useful for local development as well, more or less resolving #16550. That's great!

**Comment by czoido** [comment]
Close in favor of https://github.com/conan-io/conan/pull/17094

**Pull request closed without merging**